### PR TITLE
Start logging dicts as grouped logs

### PR DESCRIPTION
**Pull request by volkangurel**
Start logging dicts as grouped logs

### CODE REVIEW

**Review by cihati — COMMENTED** [review]
I'll take a deeper look at this, but can you update the docs as well? This is changing existing behavior.

File: layer/logged_data/log_data_runner.py [hunk 1]
```diff
@@ -114,8 +114,11 @@ def log(  # pylint: disable=too-many-statements
             elif LogDataRunner._is_tabular_dict(value):
                 if TYPE_CHECKING:
                     assert isinstance(value, dict)
-                dataframe = LogDataRunner._convert_dict_to_dataframe(value)
-                self._log_dataframe(value=dataframe, **kwargs)
+                # expand the dict into a group of logs
+                for item_key, item_value in value.items():
+                    self.log(
+                        data={item_key: item_value}, category=category, group_tag=tag
+                    )
```
Comment (cihati) [inline layer/logged_data/log_data_runner.py:121]: @mecevit this PR reverts the behavior from https://github.com/layerai/sdk/pull/92 -- are you cool with that?

Comment (mecevit) [reply layer/logged_data/log_data_runner.py:121]: There is no change on the UI or the SDK side, it's just backend. So I'm fine

Comment (cihati) [reply layer/logged_data/log_data_runner.py:121]: No, I think there's a change in _both_ UI and SDK side.

**Pull request merged**